includes/admin/class-wp-job-manager-settings.php

```diff
@@ -934,4 +956,131 @@ protected function input_multi( $option, $ignored_attributes, $values, $ignored_
+		$option['placeholder'] = esc_html__( 'Everyone (Public)', 'wp-job-manager' );
+
+		?>
+		<select
+			id="setting-<?php echo esc_attr( $option['name'] ); ?>"
+			class="regular-text settings-role-select"
+			name="<?php echo esc_attr( $option['name'] ); ?>[]"
+			multiple="multiple"
+			data-placeholder="<?php echo esc_attr( $option['placeholder'] ); ?>"
+			<?php
+			echo implode( ' ', $attributes ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			?>
+		>
+			<?php
+			foreach ( $option['options'] as $key => $name ) {
+				echo '<option value="' . esc_attr( $key ) . '" ' . selected( in_array( $key, $value, true ) ? $key : null, $key, false ) . '>' . esc_html( $name ) . '</option>';
+			}
+			?>
+		</select>
+		<?php
+
+		if ( ! empty( $option['desc'] ) ) {
+			echo ' <p class="description">' . wp_kses_post( $option['desc'] ) . '</p>';
+		}
+	}
+
+	/**
+	 * Role settings should be saved as a comma-separated list.
+	 */
+	public function pre_process_settings_save() {
+		$screen = get_current_screen();
+
+		if ( ! $screen || 'options' !== $screen->id ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Settings save will handle the nonce check.
+		if ( ! isset( $_POST['option_page'] ) || 'job_manager' !== $_POST['option_page'] ) {
+			return;
+		}
+
+		$capabilities_fields = [
+			'job_manager_browse_job_listings_capability',
+			'job_manager_view_job_listing_capability',
+		];
+		foreach ( $capabilities_fields as $capabilities_field ) {
+			// phpcs:disable WordPress.Security.NonceVerification.Missing -- Settings save will handle the nonce check.
+			if ( isset( $_POST[ $capabilities_field ] ) ) {
+				// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Sanitized by `WP_Resume_Manager_Settings::capabilities_array_to_string()`
+				$input_capabilities_field_value = wp_unslash( $_POST[ $capabilities_field ] );
+				if ( is_array( $input_capabilities_field_value ) ) {
+					$_POST[ $capabilities_field ] = self::capabilities_array_to_string( $input_capabilities_field_value );
+				}
+			}
+			// phpcs:enable WordPress.Security.NonceVerification.Missing
+		}
+	}
+
+	/**
+	 * Convert list of capabilities and roles into array of values.
+	 *
+	 * @param string $value Comma separated list of capabilities and roles.
+	 * @return array
+	 */
+	private static function capabilities_string_to_array( $value ) {
+		return array_filter(
+			array_map(
+				function( $value ) {
+					return trim( sanitize_text_field( $value ) );
+				},
+				explode( ',', $value )
+			)
+		);
+	}
+
+	/**
+	 * Convert array of capabilities and roles into a comma separated list.
+	 *
+	 * @param array $value Array of capabilities and roles.
+	 * @return string
+	 */
+	private static function capabilities_array_to_string( $value ) {
+		if ( ! is_array( $value ) ) {
+			return '';
+		}
+
+		$caps = array_filter( array_map( 'sanitize_text_field', $value ) );
```

Comment:
I think you could try to check against the list of values returned by `get_capabilities_and_roles` here, just to avoid possible issues with invalid values being injected by the client. WDYT?